src/transformers/models/speech_to_text/modeling_tf_speech_to_text.py

```diff
@@ -779,7 +779,7 @@ def _get_feat_extract_output_lengths(self, input_lengths: tf.Tensor):
     def _get_feature_vector_attention_mask(self, feature_vector_length, attention_mask):
         # generate creates 3D attention mask, because of the shape of input_features
         # convert it to 2D if thats the case
-        if len(attention_mask.shape) > 2:
+        if len(shape_list(attention_mask)) > 2:
```

Comment:
Same comment for `shape_list`. Here the issue is that `.shape` is not iterable in graph mode, but `.rank` should be okay.

Reply:
Updated :)